lib/lrama/counterexamples/example.rb

```diff
@@ -57,12 +58,14 @@ def _derivations(paths)
             when ProductionPath
               derivation = Derivation.new(item, derivation)
               current = :production
+            else
+              return nil
```

Comment:
If I remember correctly, `path` should be one of StartPath, TransitionPath or ProductionPath then it's better to `raise "Unexpected. #{path}"` than return nil.

We can omit `?` for some methods by this change.

```diff
diff --git a/lib/lrama/counterexamples/example.rb b/lib/lrama/counterexamples/example.rb
index 01995ab..bb08428 100644
--- a/lib/lrama/counterexamples/example.rb
+++ b/lib/lrama/counterexamples/example.rb
@@ -38,7 +38,7 @@ module Lrama
       private

       def _derivations(paths)
-        derivation = nil #: Derivation?
+        derivation = nil #: Derivation
         current = :production
         last_path = paths.last #: Path
         lookahead_sym = last_path.to.item.end_of_rule? ? @conflict_symbol : nil
@@ -59,7 +59,7 @@ module Lrama
               derivation = Derivation.new(item, derivation)
               current = :production
             else
-              return nil
+              raise "Unexpected. #{path}"
             end

             if lookahead_sym && item.next_next_sym && item.next_next_sym.first_set.include?(lookahead_sym)
diff --git a/sig/lrama/counterexamples/example.rbs b/sig/lrama/counterexamples/example.rbs
index ba55d46..fae1a98 100644
--- a/sig/lrama/counterexamples/example.rbs
+++ b/sig/lrama/counterexamples/example.rbs
@@ -31,13 +31,13 @@ module Lrama

       def path2_item: () -> States::Item

-      def derivations1: () -> Derivation?
+      def derivations1: () -> Derivation

-      def derivations2: () -> Derivation?
+      def derivations2: () -> Derivation

       private

-      def _derivations: (::Array[StartPath | TransitionPath | ProductionPath] paths) -> Derivation?
+      def _derivations: (::Array[StartPath | TransitionPath | ProductionPath] paths) -> Derivation

       def find_derivation_for_symbol: (StateItem state_item, Grammar::Symbol sym) -> Derivation?
     end
```

Reply:
I have set you as co-author and committed. Thanks for pointing that out.